### PR TITLE
22 eatgads output

### PR DESCRIPTION
**Pull request by sachseka**
please review implementation of new function prep2gads

### CODE REVIEW

File: R/prep2gads.R [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  labels2 <- dplyr::full_join(values3, labels1, by="varName")
+  labels2 <- labels2[,c("varName", "varLabel", "format", "display_width", "labeled", "value", "valLabel", "missings")]
+  labels2$labeled <- ifelse(!is.na(labels2$varLabel) | !is.na(labels2$valLabel), "yes", "no")
+  suppressWarnings(labels2 <- collapseMissings(labels2, missing.rule = misTypes, standard=FALSE))
+
+  dat2 <- collapseMissings(dat, missing.rule = misTypes, standard=FALSE)
+  suppressWarnings(dat2 <- eatTools::asNumericIfPossible(dat2, force.string=FALSE))
+
+  labels2 <- labels2[labels2$varName %in% names(dat2),]
```
Comment (franikowsp) [inline R/prep2gads.R:82]: Minor nit: Could the names be changed to something more declarative, e.g., `labels_joined` or `dat_collapsed`? 🤔

**Review by franikowsp — COMMENTED** [review]
This seems to be a very useful expansion to the `eatPrep` package. Every addition was easy to understand (and therefore, I added some ideas) and the amount of code to review was well manageable. 😊

File: R/prep2gads.R [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  } else {
+    values2 <- values[!duplicated(paste0(values$unit,values$value)),]
+  }
+
+  # restructure
+  if(trafoType == "scored") {
+     values3 <- data.frame(varName=values2$unit, value=values2$valueRecode, valLabel=values2$valueLabel, missings=ifelse(grepl("^m",values2$valueType, ignore.case=TRUE),"miss","valid"))
+  } else {
+    values3 <- data.frame(varName=values2$unit, value=values2$value, valLabel=values2$valueLabel, missings=ifelse(grepl("^m",values2$valueType, ignore.case=TRUE),"miss","valid"))
+  }
```
[the remark this answers is not in the document]
Comment (sachseka) [reply R/prep2gads.R:72]: since values2 is needed, there would be no gain I think

File: R/prep2gads.R (hunk 1 above)
Comment (sachseka) [reply R/prep2gads.R:82]: difficult to capture all transformations (joined, sorted, collapsed, ..) in on suffix; i prefer it as is

**Pull request closed without merging**